controllers/settings.strorefronts.js

```diff
@@ -10,6 +11,60 @@ const get = async (req, res) => {
   return res.status(HTTPStatus.OK).send(settings.values)
 }
 
+const put = async (req, res) => {
+  if (!req.body.name || !req.body.payload) {
+    return res.status(HTTPStatus.BAD_REQUEST).send({ errors: ['name and payload fields not exist. Are mandatory fields.'] })
+  }
+  const settings = await models.SettingsStoreFronts.findOne({ where: { name: req.body.name } })
+  if (settings) {
+    const dbOperationUpdate = await sequelize.transaction()
+    try {
+      const [rowsUpdated, [updated]] = await models.SettingsStoreFronts.update(
+        {
+          values: req.body.payload,
+        },
+        {
+          where: { name: req.body.name },
+          returning: true,
+        },
+      )
+      if (!rowsUpdated) {
+        await dbOperationUpdate.rollback()
+        return res.status(HTTPStatus.BAD_GATEWAY).send({ errors: ['Failed to Update Settings'] })
+      }
+      if (updated) {
+        return res.status(HTTPStatus.OK).send({ errors: ['Settings Updated'] })
+      }
+    } catch (error) {
+      if (dbOperationUpdate) await dbOperationUpdate.rollback()
+      log.error(error, '[UPDATE STOREFRONTS SETTINGS]')
+      return res.status(HTTPStatus.BAD_GATEWAY).send({ errors: ['Failed to Update Settings'] })
+    }
+  }
+
+  const dbOperationInsert = await sequelize.transaction()
+  try {
+    const insert = await models.SettingsStoreFronts.create(
```

Comment:
Instead of a block for update and another for insert, why not:
- find
- if not found (empty), insert
- otherwise, update
See controllers/settings.js:35 for an example